Modules/xSharePoint/DSCResources/MSFT_xSPTimerJobState/MSFT_xSPTimerJobState.psm1

```diff
@@ -0,0 +1,152 @@
+            # Timer job is not associated to web application
+            return @{
+                # Set the timer job settings
+                Name = $params.Name
+                Enabled = -not $timerjob.IsDisabled
+                Schedule = $timerjob.Schedule
+                InstallAccount = $params.InstallAccount
+            }
+        } else {
+            # Timer job is associated to web application
+            return @{
+                # Set the timer job settings
+                Name = $params.Name
+                WebApplication = $timerjob.WebApplication.Name
+                Enabled = -not $timerjob.IsDisabled
+                Schedule = $timerjob.Schedule
+                InstallAccount = $params.InstallAccount
+            }
+        }
+    }
+
+    return $result
+}
+
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [parameter(Mandatory = $true)] [System.String] $Name,
+        [parameter(Mandatory = $false)] [System.String] $WebApplication,
+        [parameter(Mandatory = $false)] [System.Boolean] $Enabled,
+        [parameter(Mandatory = $false)] [System.String] $Schedule,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+    )
+
+    Write-Verbose -Message "Setting timer job settings"
+
+    Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+
+        try {
+            $spFarm = Get-SPFarm
+        } catch {
+            throw "No local SharePoint farm was detected. Timer job settings will not be applied"
+            return
+        }
+        
+        Write-Verbose -Message "Start update"
+
+        # Set the timer job settings
+        if ($params.ContainsKey("Enabled")) { 
+            # Enable/Disable timer job
+            if ($params.Enabled) {
+                Write-Verbose -Verbose "Enable timer job $($params.Name)"
+                if ($params.ContainsKey("WebApplication")) {
+                    Enable-SPTimerJob $params.Name -WebApplication $params.WebApplication
+                } else {
+                    Enable-SPTimerJob $params.Name
+                }
+            } else {
+                Write-Verbose -Verbose "Disable timer job $($params.Name)"
+                if ($params.ContainsKey("WebApplication")) {
+                    Disable-SPTimerJob $params.Name -WebApplication $params.WebApplication
+                } else {
+                    Disable-SPTimerJob $params.Name
+                }
+            }
+        }
+
+        if ($params.ContainsKey("Schedule")) {
+            # Set timer job schedule
+            Write-Verbose -Verbose "Set timer job $($params.Name) schedule"
+            if ($params.ContainsKey("WebApplication")) {
+                try {
+                    Set-SPTimerJob $params.Name -WebApplication $params.WebApplication -Schedule $params.Schedule -EA Stop
+                } catch {
```

Comment:
The catch statement here assumes that the schedule error is the only thing that could go wrong. Can we catch a specific type of exception, or catch it where the error contains certain text. That way we won't cover other exceptions with this message by accident.